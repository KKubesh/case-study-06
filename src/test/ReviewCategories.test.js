import React from 'react';
import { render, cleanup } from 'react-testing-library';
import ReviewCatagories from '../components/ProductReview/ReviewResponse/ReviewCategories';

afterEach(cleanup);

test('<ReviewCatagories />', () => {
    const { getByTestId, container } = render(<ReviewCatagories/>);
    expect(getByTestId('categories')).toBeTruthy();
    expect(getByTestId('categories-review')).toBeTruthy();
    expect(getByTestId('categories-pro').textContent).toBe('PRO');
    expect(getByTestId('categories-con').textContent).toBe('CON');
    expect(getByTestId('categories-pro-text').textContent).toBe('most helpful 4-5 star review');
    expect(getByTestId('categories-con-text').textContent).toBe('most helpful 1-2 star review');
    
    expect(container.firstChild).toMatchSnapshot();
});