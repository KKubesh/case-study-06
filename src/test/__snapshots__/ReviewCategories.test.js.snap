// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewCatagories /> 1`] = `
<div
  class="row-flex"
  data-testid="categories"
>
  <div
    class="categories-review"
    data-testid="categories-review"
  >
    <h3
      data-testid="categories-pro"
    >
      PRO
    </h3>
    <h4
      data-testid="categories-pro-text"
    >
      most helpful 4-5 star review
    </h4>
  </div>
  <div
    class="categories-review"
    data-testid="categories-review"
  >
    <h3
      data-testid="categories-con"
    >
      CON
    </h3>
    <h4
      data-testid="categories-con-text"
    >
      most helpful 1-2 star review
    </h4>
  </div>
</div>
`;
